src/mail.sh

```diff
@@ -46,10 +46,39 @@ function mail_setup()
   return 0
 }
 
+function mail_verify()
+{
+  local count=0
+  local -a missing_conf
+  local -a min_conf=('user.name' 'user.email' 'sendemail.smtpencryption'
+    'sendemail.smtpserver' 'sendemail.smtpuser' 'sendemail.smtpserverport')
+  local set_confs
+
+  # say "${min_conf[*]}"
+  set_confs=$(git config --get-regexp 'user|sendemail\.smtp')
+  # say "*****$set_confs*****"
+  for config in "${min_conf[@]}"; do
+    if ! echo "$set_confs" | grep -cF "$config" - &> /dev/null; then
+      # printf '%s\n' "$config;"
+      missing_conf[$count]="$config"
+      count=$((count + 1))
+    fi
+  done
+  # echo "COUNT: $count"
+  if [[ $count -gt 0 ]]; then
+    complain "Missing configurations needed for send-email: ${missing_conf[*]}"
+    return 1
+  fi
+  say "correct config"
```

Comment:
Do you know if git-send-email has any feature that checks config? Maybe we can try to use the `dry` option just to check if everything is fine?

Reply:
Not as far as I could tell. I tried a few tests hiding some of my configurations, and `git send-email` just tried to assume most of them and get them from the commit messages when possible. While using the `--dry-run` option the only settings it complained about were `user.name` and `user.email`. Without it it obviously failed, but it attempts to send it.